fury/tests/test_ui.py

```diff
@@ -1010,49 +1010,76 @@ def timer_callback(_obj, _event):
+
+    length_ = 200
+    multi_samples = 32
+    max_peels = 8
+
+    st_x = np.arange(length_)
+    st_y = np.sin(np.arange(length_))
+    st_z = np.zeros(st_x.shape)
+    st = np.zeros((length_, 3))
+    st[:, 0] = st_x
+    st[:, 1] = st_y
+    st[:, 2] = st_z
+
+    all_st = []
+    all_st.append(st)
+    for i in range(1000):
+        all_st.append(st + i * np.array([0., .5, 0]))
+
+    # st_actor = actor.line(all_st, linewidth=1)
+    # TODO: textblock disappears when lod=True
+    st_actor = actor.streamtube(all_st, linewidth=0.1, lod=False)
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_ring_slider_2d":
-        test_ui_ring_slider_2d(recording=False)
+    scene = window.Scene()
+    scene.background((1, 1., 1))
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_range_slider":
-        test_ui_range_slider(interactive=False)
+    # quick game style antialiasing
+    scene.fxaa_on()
+    scene.fxaa_off()
+    # the good staff is later with multi-sampling
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_option":
-        test_ui_option(interactive=False)
+    tb = ui.TextBlock2D(font_size=40, color=(1, 0.5, 0))
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_checkbox":
-        test_ui_checkbox(interactive=False)
+    panel = ui.Panel2D(position=(400, 400), size=(400, 400))
+    panel.add_element(tb, (0.2, 0.5))
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_radio_button":
-        test_ui_radio_button(interactive=False)
+    counter = itertools.count()
+    showm = window.ShowManager(scene,
+                               size=(1980, 1080), reset_camera=False,
+                               order_transparent=True,
+                               multi_samples=multi_samples,
+                               max_peels=max_peels,
+                               occlusion_ratio=0.0)
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_listbox_2d":
-        test_ui_listbox_2d(interactive=False)
+    showm.initialize()
+    scene.add(panel)
+    scene.add(st_actor)
+    scene.reset_camera_tight()
+    scene.zoom(5)
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_image_container_2d":
-        test_ui_image_container_2d(interactive=False)
+    def timer_callback(_obj, _event):
+        cnt = next(counter)
+        # tb.message = "Let's count to 10 and exit :" + str(cnt)
+        if cnt % 5 == 0:
+            msg = "FPS " + str(np.round(scene.frame_rate, 0)) + ' ' + str(cnt)
+            tb.message = msg
+            showm.render()
+        if cnt > 10:
+            showm.exit()
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_timer":
-        test_timer()
+    # Run every 200 milliseconds
+    showm.add_timer_callback(True, 200, timer_callback)
+    showm.start()
 
-    if len(sys.argv) <= 1 or sys.argv[1] == "test_ui_file_menu_2d":
-        test_ui_file_menu_2d(interactive=False)
+    arr = window.snapshot(scene, offscreen=True)
+    npt.assert_(np.sum(arr) > 0)
```

Comment:
In this test, I do not see where your test the frame rate and the anti-aliasing. Can you tell me more?